src/handlers/assign.rs

```diff
@@ -693,14 +765,57 @@ fn find_reviewer_from_names(
     //
     // These are all ideas for improving the selection here. However, I'm not
     // sure they are really worth the effort.
-    Ok(candidates
+
+    // filter out team members without capacity
+    let filtered_candidates = filter_by_capacity(db, &candidates)
+        .await
+        .expect("Error while filtering out team members");
+
+    if filtered_candidates.is_empty() {
+        return Err(FindReviewerError::AllReviewersFiltered {
+            initial: names.to_vec(),
+            filtered: names.to_vec(),
+        });
+    }
+
+    log::debug!("Filtered list of candidates: {:?}", filtered_candidates);
+
+    Ok(filtered_candidates
```

Comment:
Quickly skimming on my phone. I'm curious how we currently users on vacation? I don't see any logic here or in the function below (but I may be missing it).

Reply:
Users on vacation are listed in the [triagebot file](https://github.com/rust-lang/rust/blob/ca7d34efa94afe271accf2bd3d44152a5bd6fff1/triagebot.toml#L771) and then filtered out [here](https://github.com/rust-lang/triagebot/blob/c6fa165f057475e34d5e968c89c25b8962b42a43/src/handlers/assign.rs#L726-L738), just right before (inside `candidate_reviewers_from_names`). Then we apply a second pass and filter out team members after reading their work queue.

My plan is to deprecate the "on vacation" thing altogether and revert the patch that introduced it (https://togithub.com/rust-lang/triagebot/pull/1712). To clarify: the "on vacation" feature is sound but should be handled in a more fine-grained way